src/librustc_mir/transform/simplify_branches.rs

```diff
@@ -30,26 +30,30 @@ impl<'l, 'tcx> MirPass<'tcx> for SimplifyBranches<'l> {
         for block in mir.basic_blocks_mut() {
             let terminator = block.terminator_mut();
             terminator.kind = match terminator.kind {
-                // TerminatorKind::If { ref targets, cond: Operand::Constant(Constant {
-                //     literal: Literal::Value {
-                //         value: ConstVal::Bool(cond)
-                //     }, ..
-                // }) } => {
-                //     if cond {
-                //         TerminatorKind::Goto { target: targets.0 }
-                //     } else {
-                //         TerminatorKind::Goto { target: targets.1 }
-                //     }
-                // }
-
+                TerminatorKind::SwitchInt { discr: Operand::Constant(Constant {
+                    literal: Literal::Value { ref value }, ..
+                }), ref values, ref targets, .. } => {
+                    if let Some(ref constint) = value.to_const_int() {
+                        let (otherwise, targets) = targets.split_last().unwrap();
```

Comment:
This code suggests the "blocks 1 longer than values invariant" is still in effect.